tests/unit/Term/AliasGroupListTest.php

```diff
@@ -170,20 +171,13 @@ public function testGivenInvalidLanguageCode_removeByLanguageThrowsException( $l
 	}
 
 	public function testGivenEmptyGroups_constructorRemovesThem() {
-		$enGroup = new AliasGroup( 'en', array( 'foo' ) );
-
 		$list = new AliasGroupList( array(
 			new AliasGroup( 'de' ),
-			$enGroup,
-			new AliasGroup( 'en' ),
-			new AliasGroup( 'nl' ),
-		) );
-
-		$expectedList = new AliasGroupList( array(
+			new AliasGroup( 'en', array( 'foo' ) ),
 			new AliasGroup( 'en' ),
 		) );
 
-		$this->assertEquals( $expectedList, $list );
+		$this->assertEquals( new AliasGroupList(), $list );
```

Comment:
This test is the whole reason why I touched this code. Multiple issues:

1. `$expectedList` was empty, because it was constructed with an empty group. This means the assertion compared an empty list with an empty list. This patch does not change that.
2. Sure, we could add a non-empty group to both constructors. Again, multiple issues with that:
 1. There are many other tests that already check if the constructor accepts non-empty groups. We do not need to repeat this assertion again, in a test that's about something else.
 2. The assertion would compare code with itself. If one constructors stops accepting non-empty groups, the other would do the same and the assertion would succeed. Which is, by the way, an other reason why I believe comparing objects with `assertEquals` is bogus. A test that compares the results of two constructors (or two identical method calls) with `assertEquals` is not worth much because of this issue. An assertion using `isEmpty` or `count` would be much more reliable in this specific example.